### PR TITLE
Organizer waitlist

### LINKED ISSUE #19: US 02.02.01 O - WL - View List of Entrants on Event Waiting List
An event organizer should be able to see which entrant users have joined their event waiting list. 

Associated Features

- [x] #82

### LINKED ISSUE #27: US 02.05.03 O - WL - Redraw to Replace Applicants
Draw a replacement applicant from the pooling system when a previously selected applicant cancels or rejects the invitation. 

Associated Features
- [ ]  #51

### LINKED ISSUE #31: US 02.06.01 O - EN - View List of Invited Entrants
View a list of all chosen entrants who are invited to sign up. 

Associated Features
- [ ]  #53 
- [x] #49

### LINKED ISSUE #32: US 02.06.02 O - EN - View List of Cancelled Entrants
View a list  of all the canceled entrants --> entrants who declined invite and then organizer cancelled. 

Associated Features
- [ ]  #53 
- [x] #49

### LINKED ISSUE #33: US 02.06.04 O - EN - Cancel Entrants that Decline Invite
If an invited entrant cancels and event, organizer can "cancel" them. 

Associated Features

- [x] #54

### LINKED ISSUE #52: US 02.06.03 O - EN - View Final List of attendess
Organizers can see who is expected to attend the event. ( People who have  accepted the invite)

 Associated Features 
- [x] #53 
- [x] #49

### LINKED ISSUE #53: Organizer - View Entrant Lists
Organizers need to be able to view entrant based on their statuses for certain events #49. 
Should have the ability to view 

- [ ] invited entrants #31 
- [ ] cancelled entrants #32 
- [ ] final list of entrants #52 

For this:
- Query the db to return the entrants who have the specific status when the "View List of [Status name]" is clicked. 
- On the view age a list of these entrants should be visible 
- Real time updates
- Error messages: If none have been chosen or if db query failed (error handling/validation). 

Associated Features

- [x] #64 
- [x] #49

### LINKED ISSUE #54: Organizer - Cancel Entrants
If a user is invited to an vent off the waiting list but then declines, organizers need the ability  to cancel them. 

Need a cancel button for each declined invitee or entrants that did not sign when asked to , When cancelled, update status to cancelled, in db #49 

Validation: Can only cancel declined invitees. 

Associated User  Stories

- [ ]  #33 
- [x] #49

### PR DESCRIPTION
**Pull request by timy-13**
Create waitlist view for organizer

### CODE REVIEW

File: app/src/main/java/com/example/nachosbusiness/events/Event.java [hunk 1]
```diff
@@ -1,11 +1,12 @@
 package com.example.nachosbusiness.events;
 
-import com.example.nachosbusiness.utils.QRUtil;
```
Comment (aabrady6) [inline app/src/main/java/com/example/nachosbusiness/events/Event.java]: This needs to import from utils.qrutil

File: app/src/main/java/com/example/nachosbusiness/organizer_views/WaitlistFragment.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        });
+
+        ImageButton menuButton = view.findViewById(R.id.menu);
+        menuButton.setOnClickListener(v -> {
+
+        });
+
+        ImageButton backButton = view.findViewById(R.id.back);
+        backButton.setOnClickListener(v -> {
+            Intent intent = new Intent(getActivity(), Dashboard.class);
```
Comment (aabrady6) [inline app/src/main/java/com/example/nachosbusiness/organizer_views/WaitlistFragment.java]: App crashes here

File: app/src/main/java/com/example/nachosbusiness/organizer_views/WaitlistFragment.java [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                    }
+
+                    int count = Integer.parseInt(countInput.getText().toString());
+                    if (count > listManager.getWaitList().size())
+                    {
+                        Toast.makeText(getContext(), "Sample cannot be greater than number of entrants in waitlist!", Toast.LENGTH_SHORT).show();
+                    } else if (count < 0) {
+                        Toast.makeText(getContext(), "Sample must be greater than zero!", Toast.LENGTH_SHORT).show();
+                    } else {
+                        sampleCount = count;
```
Comment (aabrady6) [inline app/src/main/java/com/example/nachosbusiness/organizer_views/WaitlistFragment.java]: App crashes if no value is put in. Prob make this a required field.

File: app/src/main/java/com/example/nachosbusiness/organizer_views/WaitlistFragment.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        builder.setView(countInput)
+                .setTitle("Sample Entrants")
+                .setPositiveButton("Ok", (dialog, which) ->{
+                    if (listManager == null)
+                    {
+                        Toast.makeText(getContext(), "Failed to load entrants", Toast.LENGTH_SHORT).show();
+                        return;
+                    }
+
+                    int count = Integer.parseInt(countInput.getText().toString());
```
Comment (aabrady6) [inline app/src/main/java/com/example/nachosbusiness/organizer_views/WaitlistFragment.java]: maybe add some limits on input values here? Can crash app if put in big number like seth did on the demo

**Comment by timy-13** [comment]
Recent changes:
* Fixed the weird list db bug that aaron discovered
* Updated UI to show profile pictures and look a bit nicer
* Added radio buttons to choose which lists to display
* Changed the sampling to use the value predetermined at event creation

**Pull request merged**